Source/JavaScriptCore/builtins/ArrayPrototype.js

```diff
@@ -402,7 +402,8 @@ function concat(first)
 
     if (@argumentCount() === 0
         && @isJSArray(this)
-        && @tryGetByIdWithWellKnownSymbol(this, "isConcatSpreadable") === @undefined) {
+        && @tryGetByIdWithWellKnownSymbol(this, "isConcatSpreadable") === @undefined
+        && @arraySpeciesWatchpointIsValid(this)) {
 
         var result = @arrayFromFastFillWithEmpty(@Array, this);
```

Comment:
Why not checking species status in arrayFromFastFillWithEmpty function?

Reply:
Because I thought arrayFromFastFillWithEmpty may also be used to implement fast paths for other functions, and such other functions may not need to do the species check.